# Overview
###### The Nobel Prize is perhaps the world's most well known scientific award. Every year it is given to scientists and scholars in chemistry, literature, physics, medicine, economics, and peace. The first Nobel Prize was handed out in 1901, and at that time the prize was Eurocentric and male-focused, but nowadays it's not biased in any way. Surely, right? Well, we’re going to find out! The Nobel Foundation has made a dataset available of all prize winners from the start of the prize, in 1901, to 2016. Let’s load it in and check it out.

## Imports

In [2]:
import numpy as np
import os
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

KeyboardInterrupt: 

## Reading Data

In [3]:

dff = pd.read_csv('archive.csv')
df = dff.copy()
df.sample(5)

Year    Category                                              Prize  \
446  1973   Economics  The Sveriges Riksbank Prize in Economic Scienc...   
278  1952       Peace                         The Nobel Peace Prize 1952   
606  1988  Literature                 The Nobel Prize in Literature 1988   
675  1995   Chemistry                  The Nobel Prize in Chemistry 1995   
782  2003     Physics                    The Nobel Prize in Physics 2003   

                                            Motivation Prize Share  \
446  "for the development of the input-output metho...         1/1   
278                                                NaN         1/1   
606  "who, through works rich in nuance - now clear...         1/1   
675  "for their work in atmospheric chemistry, part...         1/3   
782  "for pioneering contributions to the theory of...         1/3   

     Laureate ID Laureate Type            Full Name  Birth Date  \
446          683    Individual     Wassily Leontief  1906-08-05   
278          513    Individual    Albert Schweitzer  1875-01-14   
606          665    Individual       Naguib Mahfouz  1911-12-11   
675          283    Individual  F. Sherwood Rowland  1927-06-28   
782          766    Individual  Alexei A. Abrikosov  1928-06-25   

         Birth City                                 Birth Country   Sex  \
446  St. Petersburg                                        Russia  Male   
278     Kaysersberg                              Germany (France)  Male   
606           Cairo                                         Egypt  Male   
675    Delaware, OH                      United States of America  Male   
782          Moscow  Union of Soviet Socialist Republics (Russia)  Male   

               Organization Name Organization City      Organization Country  \
446           Harvard University     Cambridge, MA  United States of America   
278                          NaN               NaN                       NaN   
606                          NaN               NaN                       NaN   
675     University of California        Irvine, CA  United States of America   
782  Argonne National Laboratory       Argonne, IL  United States of America   

     Death Date          Death City             Death Country  
446  1999-02-05        New York, NY  United States of America  
278  1965-09-04           Lambaréné                     Gabon  
606  2006-08-30               Cairo                     Egypt  
675  2012-03-10  Corona del Mar, CA  United States of America  
782         NaN                 NaN                       NaN

## Data Exploration and Cleaning

### How many nobel prizes we have between 1901 and 2016

In [4]:
print("Shape\n%s Rows \n%s Columns " % (df.shape[0],df.shape[1]))

Shape
969 Rows 
18 Columns 


### How many male nobel prizes and how many female nobel prizes

In [5]:
df1 = df.groupby(['Sex']).size()
print (df1)

Sex
Female     50
Male      893
dtype: int64


## Different nationalities winning the nobel prize

In [6]:
df2 =  df.groupby("Birth Country").agg('size')

print(df2.sort_values( ascending=False))


Birth Country
United States of America                  276
United Kingdom                             88
Germany                                    70
France                                     53
Sweden                                     30
                                         ... 
Ottoman Empire (Republic of Macedonia)      1
Ottoman Empire (Turkey)                     1
Pakistan                                    1
Persia (Iran)                               1
Java, Dutch East Indies (Indonesia)         1
Length: 121, dtype: int64


### From the previous group by we can conclude that the USA dominates in the noble prizes birth country as there is 276 nobel prize holder from USA

### Calculate the proption of female lauretes per decade

In [17]:
df3=df[df["Sex"] == "Female"].groupby("Year").size()
print(df3)

Year
1903    1
1905    1
1909    1
1911    1
1926    1
1928    1
1931    1
1935    1
1938    1
1945    1
1946    1
1947    1
1963    1
1964    1
1966    1
1976    2
1977    1
1979    1
1982    1
1983    1
1986    1
1988    1
1991    2
1992    1
1993    1
1995    1
1996    1
1997    1
2003    1
2004    3
2007    1
2008    1
2009    6
2011    3
2013    1
2014    2
2015    2
dtype: int64


### Calculate the proption of female lauretes per category

In [19]:
df4=df[df["Sex"] == "Female"].groupby("Category").size()
print(df4)

Category
Chemistry      4
Economics      2
Literature    14
Medicine      12
Peace         16
Physics        2
dtype: int64


### The first woman to win the Nobel Prize


In [33]:
s=df[df["Sex"] == "Female"]
s2=s.sort_values(by=['Year'])
s3=s2.nsmallest(1,'Year')
s3

Year Category                            Prize  \
19  1903  Physics  The Nobel Prize in Physics 1903   

                                           Motivation Prize Share  \
19  "in recognition of the extraordinary services ...         1/4   

    Laureate ID Laureate Type                    Full Name  Birth Date  \
19            6    Individual  Marie Curie, née Sklodowska  1867-11-07   

   Birth City            Birth Country     Sex Organization Name  \
19     Warsaw  Russian Empire (Poland)  Female               NaN   

   Organization City Organization Country  Death Date  Death City  \
19               NaN                  NaN  1934-07-04  Sallanches   

   Death Country  
19        France

### Who won the nobel prize more than once?

In [39]:
df5=  df.groupby("Full Name").agg('size')
print(df5.sort_values( ascending=False))


Full Name
Jack W. Szostak                                                                      3
Comité international de la Croix Rouge (International Committee of the Red Cross)    3
Randy W. Schekman                                                                    2
Robert J. Lefkowitz                                                                  2
Roderick MacKinnon                                                                   2
                                                                                    ..
National Dialogue Quartet                                                            1
Naguib Mahfouz                                                                       1
Nadine Gordimer                                                                      1
Médecins Sans Frontières                                                             1
A. Michael Spence                                                                    1
Length: 904, dtype: int64


### To Know the number of entries and the columns

In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 969 entries, 0 to 968
Data columns (total 18 columns):
Year                    969 non-null int64
Category                969 non-null object
Prize                   969 non-null object
Motivation              881 non-null object
Prize Share             969 non-null object
Laureate ID             969 non-null int64
Laureate Type           969 non-null object
Full Name               969 non-null object
Birth Date              940 non-null object
Birth City              941 non-null object
Birth Country           943 non-null object
Sex                     943 non-null object
Organization Name       722 non-null object
Organization City       716 non-null object
Organization Country    716 non-null object
Death Date              617 non-null object
Death City              599 non-null object
Death Country           605 non-null object
dtypes: int64(2), object(16)
memory usage: 136.4+ KB


### Checking how many null values for each column

In [49]:
df.isnull().sum()

Year                      0
Category                  0
Prize                     0
Motivation               88
Prize Share               0
Laureate ID               0
Laureate Type             0
Full Name                 0
Birth Date               29
Birth City               28
Birth Country            26
Sex                      26
Organization Name       247
Organization City       253
Organization Country    253
Death Date              352
Death City              370
Death Country           364
dtype: int64

### What are the values for the Laureate Type

In [50]:
pd.unique(df["Laureate Type"])

array(['Individual', 'Organization'], dtype=object)

### Number of Cells having Laureate type = Organization

In [51]:
df[df["Laureate Type"] == "Organization"].shape

(30, 18)

In [53]:
df[df["Sex"].isnull()]

Year Category                       Prize  \
24   1904    Peace  The Nobel Peace Prize 1904   
61   1910    Peace  The Nobel Peace Prize 1910   
90   1917    Peace  The Nobel Peace Prize 1917   
206  1938    Peace  The Nobel Peace Prize 1938   
222  1944    Peace  The Nobel Peace Prize 1944   
244  1947    Peace  The Nobel Peace Prize 1947   
245  1947    Peace  The Nobel Peace Prize 1947   
295  1954    Peace  The Nobel Peace Prize 1954   
365  1963    Peace  The Nobel Peace Prize 1963   
366  1963    Peace  The Nobel Peace Prize 1963   
383  1965    Peace  The Nobel Peace Prize 1965   
416  1969    Peace  The Nobel Peace Prize 1969   
498  1977    Peace  The Nobel Peace Prize 1977   
543  1981    Peace  The Nobel Peace Prize 1981   
578  1985    Peace  The Nobel Peace Prize 1985   
610  1988    Peace  The Nobel Peace Prize 1988   
682  1995    Peace  The Nobel Peace Prize 1995   
705  1997    Peace  The Nobel Peace Prize 1997   
728  1999    Peace  The Nobel Peace Prize 1999   
754  2001    Peace  The Nobel Peace Prize 2001   
808  2005    Peace  The Nobel Peace Prize 2005   
821  2006    Peace  The Nobel Peace Prize 2006   
833  2007    Peace  The Nobel Peace Prize 2007   
909  2012    Peace  The Nobel Peace Prize 2012   
927  2013    Peace  The Nobel Peace Prize 2013   
955  2015    Peace  The Nobel Peace Prize 2015   

                                            Motivation Prize Share  \
24                                                 NaN         1/1   
61                                                 NaN         1/1   
90                                                 NaN         1/1   
206                                                NaN         1/1   
222                                                NaN         1/1   
244                                                NaN         1/2   
245                                                NaN         1/2   
295                                                NaN         1/1   
365                                                NaN         1/2   
366                                                NaN         1/2   
383                                                NaN         1/1   
416                                                NaN         1/1   
498                                                NaN         1/1   
543                                                NaN         1/1   
578                                                NaN         1/1   
610                                                NaN         1/1   
682  "for their efforts to diminish the part played...         1/2   
705  "for their work for the banning and clearing o...         1/2   
728  "in recognition of the organization's pioneeri...         1/1   
754  "for their work for a better organized and mor...         1/2   
808  "for their efforts to prevent nuclear energy f...         1/2   
821  "for their efforts to create economic and soci...         1/2   
833  "for their efforts to build up and disseminate...         1/2   
909  "for over six decades contributed to the advan...         1/1   
927  "for its extensive efforts to eliminate chemic...         1/1   
955  "for its decisive contribution to the building...         1/1   

     Laureate ID Laureate Type  \
24           467  Organization   
61           477  Organization   
90           482  Organization   
206          503  Organization   
222          482  Organization   
244          508  Organization   
245          509  Organization   
295          515  Organization   
365          482  Organization   
366          523  Organization   
383          525  Organization   
416          527  Organization   
498          537  Organization   
543          515  Organization   
578          547  Organization   
610          550  Organization   
682          561  Organization   
705          564  Organization   
728          568  Organization   
754          748  Organization   
808          797  Organization   
821          81

In [55]:
#df[df["Prize Share"] == "1/2"]
df[df["Full Name"] == "Paul Ehrlich"]

Year  Category                                           Prize  \
46  1908  Medicine  The Nobel Prize in Physiology or Medicine 1908   
47  1908  Medicine  The Nobel Prize in Physiology or Medicine 1908   

                                    Motivation Prize Share  Laureate ID  \
46  "in recognition of their work on immunity"         1/2          302   
47  "in recognition of their work on immunity"         1/2          302   

   Laureate Type     Full Name  Birth Date           Birth City  \
46    Individual  Paul Ehrlich  1854-03-14  Strehlen (Strzelin)   
47    Individual  Paul Ehrlich  1854-03-14  Strehlen (Strzelin)   

       Birth Country   Sex                                  Organization Name  \
46  Prussia (Poland)  Male                              Goettingen University   
47  Prussia (Poland)  Male  Königliches Institut für experimentelle Therap...   

        Organization City Organization Country  Death Date  \
46              Göttingen              Germany  1915-08-20   
47  Frankfurt-on-the-Main              Germany  1915-08-20   

                  Death City Death Country  
46  Bad Homburg vor der Höhe       Germany  
47  Bad Homburg vor der Höhe       Germany

In [57]:

#df[df["Organization Name"].isnull()]
l = df[df["Organization Name"].isnull()]
k = l[l["Laureate Type"] == "Organization"]
print("Shape\n%s Rows \n%s Columns " % (k.shape[0],k.shape[1]))


Shape
30 Rows 
18 Columns 


## Cleaning 

### Removing Duplicates 

In [58]:
print(len(df))
df.drop_duplicates(subset = "Laureate ID",keep = "first", inplace = True)
print(len(df))
df[df["Full Name"] == "Paul Ehrlich"]

969
904


Year  Category                                           Prize  \
46  1908  Medicine  The Nobel Prize in Physiology or Medicine 1908   

                                    Motivation Prize Share  Laureate ID  \
46  "in recognition of their work on immunity"         1/2          302   

   Laureate Type     Full Name  Birth Date           Birth City  \
46    Individual  Paul Ehrlich  1854-03-14  Strehlen (Strzelin)   

       Birth Country   Sex      Organization Name Organization City  \
46  Prussia (Poland)  Male  Goettingen University         Göttingen   

   Organization Country  Death Date                Death City Death Country  
46              Germany  1915-08-20  Bad Homburg vor der Höhe       Germany

### Deleting Motivation values

In [59]:
del df['Motivation']
df.sample(5)

Year   Category                              Prize Prize Share  \
572  1985  Chemistry  The Nobel Prize in Chemistry 1985         1/2   
432  1971    Physics    The Nobel Prize in Physics 1971         1/1   
238  1946    Physics    The Nobel Prize in Physics 1946         1/1   
545  1981    Physics    The Nobel Prize in Physics 1981         1/4   
126  1925    Physics    The Nobel Prize in Physics 1925         1/2   

     Laureate ID Laureate Type                Full Name  Birth Date  \
572          262    Individual      Herbert A. Hauptman  1917-02-14   
432           93    Individual             Dennis Gabor  1900-06-05   
238           51    Individual  Percy Williams Bridgman  1882-04-21   
545          119    Individual  Arthur Leonard Schawlow  1921-05-05   
126           30    Individual             James Franck  1882-08-26   

          Birth City             Birth Country   Sex  \
572     New York, NY  United States of America  Male   
432         Budapest                   Hungary  Male   
238    Cambridge, MA  United States of America  Male   
545  Mount Verno, NY  United States of America  Male   
126          Hamburg                   Germany  Male   

                     Organization Name Organization City  \
572  The Medical Foundation of Buffalo       Buffalo, NY   
432                   Imperial College            London   
238                 Harvard University     Cambridge, MA   
545                Stanford University      Stanford, CA   
126              Goettingen University         Göttingen   

         Organization Country  Death Date     Death City  \
572  United States of America  2011-10-23    Buffalo, NY   
432            United Kingdom  1979-02-08         London   
238  United States of America  1961-08-20   Randolph, NH   
545  United States of America  1999-04-28  Palo Alto, CA   
126                   Germany  1964-05-21      Göttingen   

                Death Country  
572  United States of America  
432            United Kingdom  
238  United States of America  
545  United States of America  
126    West Germany (Germany)

### Fixing Incorrect Data of Laureate Type ( Org -> Individual )

In [61]:

#df.loc[df["Sex"].isnull(), "Sex"] = "Org"

mask = (df["Laureate Type"] == "Organization") & (df["Birth Date"].notnull())
df["Laureate Type"][mask] = "Individual"

l = df[df["Laureate Type"] == "Organization"]
k = l[l["Birth Date"].notnull()]
k

C:\Users\Administrator\Anaconda3\lib\site-packages\ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.


Empty DataFrame
Columns: [Year, Category, Prize, Prize Share, Laureate ID, Laureate Type, Full Name, Birth Date, Birth City, Birth Country, Sex, Organization Name, Organization City, Organization Country, Death Date, Death City, Death Country]
Index: []

In [62]:
k = df[df["Laureate Type"] == "Organization"]
print("Shape\n%s Rows \n%s Columns " % (k.shape[0],k.shape[1]))
k
#df.isnull().sum()

Shape
23 Rows 
17 Columns 


Year Category                       Prize Prize Share  Laureate ID  \
24   1904    Peace  The Nobel Peace Prize 1904         1/1          467   
61   1910    Peace  The Nobel Peace Prize 1910         1/1          477   
90   1917    Peace  The Nobel Peace Prize 1917         1/1          482   
206  1938    Peace  The Nobel Peace Prize 1938         1/1          503   
244  1947    Peace  The Nobel Peace Prize 1947         1/2          508   
245  1947    Peace  The Nobel Peace Prize 1947         1/2          509   
295  1954    Peace  The Nobel Peace Prize 1954         1/1          515   
366  1963    Peace  The Nobel Peace Prize 1963         1/2          523   
383  1965    Peace  The Nobel Peace Prize 1965         1/1          525   
416  1969    Peace  The Nobel Peace Prize 1969         1/1          527   
498  1977    Peace  The Nobel Peace Prize 1977         1/1          537   
578  1985    Peace  The Nobel Peace Prize 1985         1/1          547   
610  1988    Peace  The Nobel Peace Prize 1988         1/1          550   
682  1995    Peace  The Nobel Peace Prize 1995         1/2          561   
705  1997    Peace  The Nobel Peace Prize 1997         1/2          564   
728  1999    Peace  The Nobel Peace Prize 1999         1/1          568   
754  2001    Peace  The Nobel Peace Prize 2001         1/2          748   
808  2005    Peace  The Nobel Peace Prize 2005         1/2          797   
821  2006    Peace  The Nobel Peace Prize 2006         1/2          810   
833  2007    Peace  The Nobel Peace Prize 2007         1/2          818   
909  2012    Peace  The Nobel Peace Prize 2012         1/1          881   
927  2013    Peace  The Nobel Peace Prize 2013         1/1          893   
955  2015    Peace  The Nobel Peace Prize 2015         1/1          925   

    Laureate Type                                          Full Name  \
24   Organization  Institut de droit international (Institute of ...   
61   Organization  Bureau international permanent de la Paix (Per...   
90   Organization  Comité international de la Croix Rouge (Intern...   
206  Organization  Office international Nansen pour les Réfugiés ...   
244  Organization              Friends Service Council (The Quakers)   
245  Organization   American Friends Service Committee (The Quakers)   
295  Organization  Office of the United Nations High Commissioner...   
366  Organization  Ligue des Sociétés de la Croix-Rouge (League o...   
383  Organization            United Nations Children's Fund (UNICEF)   
416  Organization         International Labour Organization (I.L.O.)   
498  Organization                              Amnesty International   
578  Organization  International Physicians for the Prevention of...   
610  Organization                 United Nations Peacekeeping Forces   
682  Organization   Pugwash Conferences on Science and World Affairs   
705  Organization     International Campaign to Ban Landmines (ICBL)   
728  Organization                           Médecins Sans Frontières   
754  Organization                              United Nations (U.N.)   
808  Organization          International Atomic Energy Agency (IAEA)   
821  Organization                                       Grameen Bank   
833  Organization   Intergovernmental Panel on Climate Change (IPCC)   
909  Organization                                European Union (EU)   
927  Organization  Organisation for the Prohibition of Chemical W...   
955  Organization                          National Dialogue Quartet   

    Birth Date Birth City Birth Country  Sex Organization Name  \
24         NaN        NaN           NaN  NaN               NaN   
61         NaN        NaN           NaN  NaN               NaN   
90         NaN        NaN           NaN  NaN               NaN   
206        NaN        NaN           NaN  NaN               NaN   
244        NaN        NaN           NaN  NaN               NaN   
245        NaN        NaN           NaN  NaN               NaN   
295        Na

### Dropping organization since they only represent 2.5% of the dataframe ( 23 rows out of 904 rows)

In [63]:
print("Shape\n%s Rows \n%s Columns " % (df.shape[0],df.shape[1]))
df = df[df["Laureate Type"] == "Individual"]
print("Shape\n%s Rows \n%s Columns " % (df.shape[0],df.shape[1]))
df[df["Laureate Type"] == "Individual"]
df.isnull().sum()
#df[df["Death Date"].isnull()]

Shape
904 Rows 
17 Columns 
Shape
881 Rows 
17 Columns 


Year                      0
Category                  0
Prize                     0
Prize Share               0
Laureate ID               0
Laureate Type             0
Full Name                 0
Birth Date                2
Birth City                2
Birth Country             0
Sex                       0
Organization Name       220
Organization City       218
Organization Country    218
Death Date              292
Death City              309
Death Country           303
dtype: int64

### Dropping the Death City, Death Date, Death Country as they will not affect Data visualisation in any way

In [64]:
df = df.drop("Death Date", axis=1)
df = df.drop("Death City", axis=1)
df = df.drop("Death Country", axis=1)
df.isnull().sum()


Year                      0
Category                  0
Prize                     0
Prize Share               0
Laureate ID               0
Laureate Type             0
Full Name                 0
Birth Date                2
Birth City                2
Birth Country             0
Sex                       0
Organization Name       220
Organization City       218
Organization Country    218
dtype: int64


### Adding missing birth date and city for individuals 

In [65]:
df.loc[df["Laureate ID"] == 841, "Birth Date"] = "1952-04-01" #venk
df.loc[df["Laureate ID"] == 864, "Birth Date"] = "1959-09-22" #saul
df.loc[df["Laureate ID"] == 747, "Birth City"] = "Chaguanas" # sir Vidiadhar
df.loc[df["Laureate ID"] == 855, "Birth City"] = "Changchun" # liu Xiaobo
df.isnull().sum()

Year                      0
Category                  0
Prize                     0
Prize Share               0
Laureate ID               0
Laureate Type             0
Full Name                 0
Birth Date                0
Birth City                0
Birth Country             0
Sex                       0
Organization Name       220
Organization City       218
Organization Country    218
dtype: int64

In [66]:

df.loc[df["Laureate ID"] == 6, "Organization Country"] = "self" 
df.loc[df["Laureate ID"] == 6, "Organization City"] = "self" 
df.loc[df["Laureate ID"] == 6, "Organization Name"] = "self" 

df.loc[df["Laureate ID"] == 318, "Organization Country"] = "Tunisia" 

df.loc[df["Laureate ID"] == 684, "Organization Country"] = "self" 
df.loc[df["Laureate ID"] == 684, "Organization City"] = "self" 
df.loc[df["Laureate ID"] == 684, "Organization Name"] = "self" 

df.loc[df["Laureate ID"] == 685, "Organization Country"] = "self" 
df.loc[df["Laureate ID"] == 685, "Organization City"] = "self" 
df.loc[df["Laureate ID"] == 685, "Organization Name"] = "self" 

df.loc[df["Laureate ID"] == 270, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 270, "Organization City"] = "Maryland" 

df.loc[df["Laureate ID"] == 461, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 461, "Organization City"] = "Maryland" 

df.loc[df["Laureate ID"] == 770, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 770, "Organization City"] = "Maryland"

df.loc[df["Laureate ID"] == 811, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 811, "Organization City"] = "Maryland"

df.loc[df["Laureate ID"] == 831, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 831, "Organization City"] = "Maryland" 

df.loc[df["Laureate ID"] == 842, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 842, "Organization City"] = "Maryland" 

df.loc[df["Laureate ID"] == 837, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 837, "Organization City"] = "Maryland" 

df.loc[df["Laureate ID"] == 878, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 878, "Organization City"] = "Maryland" 

df.loc[df["Laureate ID"] == 885, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 885, "Organization City"] = "Maryland" 

df.loc[df["Laureate ID"] == 886, "Organization Country"] = "USA" 
df.loc[df["Laureate ID"] == 886, "Organization City"] = "Maryland" 

k = df[df["Organization Country"].isnull()]
l = k[k["Category"] != "Peace"]
l[l["Category"] != "Literature"]

df[df["Organization Country"].isnull()].shape


# ba3d keda el hytba22a meen b2a? el peace wl literature bas
# homa el wa7ideen el hyb2o be null, then momken terunno el 3 lines el ta7t
#dool be el text el t7boha, take care el order yb2a keda 3shan law 3mlto 
#3 lines el ta7t dool homa el nas el physics wl mediccine msh ht3rfo tgebeehom tany

df.loc[df["Organization Name"].isnull(), "Organization Name"] = "Self"
df.loc[df["Organization City"].isnull(), "Organization City"] = "Self"
df.loc[df["Organization Country"].isnull(), "Organization Country"] = "Self"

df.isnull().sum()


Year                    0
Category                0
Prize                   0
Prize Share             0
Laureate ID             0
Laureate Type           0
Full Name               0
Birth Date              0
Birth City              0
Birth Country           0
Sex                     0
Organization Name       0
Organization City       0
Organization Country    0
dtype: int64

In [67]:
df.to_csv('archiveData_Cleaned.csv')